source/neuropod/multiprocess/shm_tensor.hh

```diff
@@ -186,20 +186,20 @@ public:
 
     // Load an existing tensor
     // See tensor_from_id
-    SHMNeuropodTensor(const std::vector<int64_t> &dims,
-                      std::shared_ptr<void>       block,
-                      shm_tensor *                data,
-                      const SHMBlockID &          block_id)
+    [[maybe_unused]] SHMNeuropodTensor(const std::vector<int64_t> &dims,
+                                       std::shared_ptr<void>       block,
+                                       shm_tensor *                data,
+                                       const SHMBlockID &          block_id)
         : TypedNeuropodTensor<std::string>(copy_and_strip_last_dim(dims)), write_buffer_(this->get_num_elements())
     {
         auto byte_block = stdx::make_unique<SHMNeuropodTensor<uint8_t>>(dims, std::move(block), data, block_id);
 
         auto base_ptr = byte_block->get_raw_data_ptr();
-        auto max_len  = dims[dims.size() - 1];
+        auto max_len  = static_cast<size_t>(dims[dims.size() - 1]);
```

Comment:
I think this cast is not needed and as result you don't need to cast it then at line 229 again